File: C++/035_Search Insert Position.cpp
class Solution {
public:
    int searchInsert(int A[], int n, int target) {
        if (n == 0)
            return 0;
        int l = 0, r = n - 1;
        int m;
        while (l <= r) {
            m = l + ((r - l) >> 1);
            if (A[m] == target)
                return m;
            if (A[m] > target)
                r = m - 1;
            else
                l = m + 1;
        }
        return l;
    }
};

File: Python/035_Search Insert Position.py
class Solution:
    # @param {integer[]} nums
    # @param {integer} target
    # @return {integer}
    def searchInsert(self, nums, target):
        N = len(nums)
        for i in range(0, N):
            if nums[i] >= target:
                return i
        return N